jobserver/templates/login.html

```diff
@@ -0,0 +1,16 @@
+{% extends "base.html" %}
+
+{% block content %}
+<div class="container">
+  <div class="row">
+    <div class="col-sm-12">
+      <form method="POST" action="{% url "auth-login" "github" %}?next={{ request.path }}">
```

Comment:
This `next` parameter will just bring the user back to the login page which I don't think is what we want. I think instead we need to save the original page where the login link was clicked (see below) and then use that parameter as the `next` value here.

Reply:
I've used the value of `request.GET.next` for `next`, here. Consequently, a user will be returned to the page where they clicked the "Login" link in the header, rather than to the login page. I've added a template filter, for the case where a user arrives at the login page directly. In this case, they will be returned to the homepage.

Reply:
I think this is probably fine, although technically this creates an "open redirect". I'm going to seek further opinions on the extent to which we need to care about this.